src/az390.java

```diff
@@ -9414,20 +9415,24 @@ private void process_dcc_data(){
 	}
 	dc_index = dc_index + dcc_next + 1;
 	dcc_len = dcc_text.length();
-	dcc_ascii_req = 
-		 (dcc_quote == '\'' 
-			 && (    (tz390.opt_ascii 
-				      && dc_type_sfx != 'E'
-				     )
-				  || dc_type_sfx == 'A'
-				)
-         )
-	     | dcc_quote == '"';  //RPI5 and RPI73
-	if (dc_bit_len){
-		gen_dcc_bits();
-	} else {
-		gen_dcc_bytes();
-	}
+	if (dcc_len > 0 || dc_len_explicit) {                            // #509
+		dcc_ascii_req = 
+			 (dcc_quote == '\'' 
+				 && (    (tz390.opt_ascii 
+					      && dc_type_sfx != 'E'
+					     )
+					  || dc_type_sfx == 'A'
+					)
+    	     )
+		     | dcc_quote == '"';  //RPI5 and RPI73
```

Comment:
I am confused by the use of | where || is used a couple of lines above. 
Apparently it works - but is it semantically correct in this location?

Reply:
I believe that both work. However, the | is not the proper usage. See the StackOverflow article, second answer:

https://stackoverflow.com/questions/1795808/and-and-or-in-if-statements